---
title: "Final Paper"
output: pdf_document
author: "Rob St. Louis and Sasen Cain, for the Pediatric Imaging, Neurocognition and Genetics Study*"
bibliography: bibliography.bib
toc: true
---
*Data used in preparation of this article were obtained from the Pediatric Imaging, Neurocognition and Genetics Study (PING) database (http://ping.chd.ucsd.edu). As such, the investigators within PING contributed to the design and implementation of PING and/or provided data but did not participate in analysis or writing of this report. A complete listing of PING investigators can be found at https://ping-dataportal.ucsd.edu/sharing/Authors10222012.pdf.



```{r,echo=FALSE,results='hide',message=FALSE,warning=FALSE}
library(xtable)
library(ggplot2)
library(dplyr)
library(magrittr)
library(scales)



GeneticData <-read.csv("Data/DRD2andCOMTAlleles.csv",stringsAsFactors=FALSE)
BehaveData <- read.csv("Data/PING_Behavior.csv",stringsAsFactors=FALSE)
COMTData <-read.csv("Data/COMT_RS4680.csv",stringsAsFactors=FALSE)

Data <- BehaveData
#Reordering household income
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==1,5000,NA)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==2,7500,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==3,15000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==4,25000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==5,35000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==6,45000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==7,75000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==8,125000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==9,175000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==10,225000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==11,275000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_Income2 <-ifelse(Data$FDH_3_Household_Income==12,300000,Data$FDH_3_Household_Income2)
Data$FDH_3_Household_IncomeLog <-as.numeric(log10(Data$FDH_3_Household_Income2))
BehaveData <- Data

## OK, PHX_Alcohol_LifeUse_1==1 includes healthy drinking habits in college and parental guidance on how to drink wine with dinner. 
## who gets drunk? let's use that as the substance abuse DV
first_drunk = sapply(BehaveData$PHX_Alcohol_LifeAbuse_2, as.integer)
first_drunk.tab = table(first_drunk)
sum(first_drunk.tab) - first_drunk.tab["0"] ## OK, 69 kids got drunk
## just make sure they started getting drunk AFTER they started drinking!
init_drink = sapply(BehaveData$PHX_Alcohol_Initiation_1, as.integer)
table(init_drink, first_drunk)
confused_drinkers = subset(BehaveData, (first_drunk != 0) & (init_drink != 99) & (init_drink > first_drunk))  # you started drinking at age 20 and first got drunk at age 14??? REJECTED
confusedID = confused_drinkers$SubjID


BehaveData %<>% 
  dplyr::mutate(got_drunk =ifelse(PHX_Alcohol_LifeAbuse_2=="",0,
                           ifelse(substr(PHX_Alcohol_LifeAbuse_2,1,1)==0,0,1)))
## verify that nobody got put somewhere strange
table(BehaveData$got_drunk,BehaveData$PHX_Alcohol_LifeAbuse_2)

Druggies <- BehaveData %>%
  dplyr::filter(  got_drunk == 1             # we created            
         | PHX_Tobacco_Status_1==1
         | PHX_Substance_LifeUse_5==1  # marijuana
         | PHX_Substance_Coc_1==1    # crack or cocaine
         | PHX_Substance_Stims_1==1
         | PHX_Substance_Meth_1==1
         | PHX_Substance_Sedtv_1==1
         | PHX_Substance_Tranq_1==1
         | PHX_Substance_Painkiller_1==1
         | PHX_Substance_Halluc_1==1
         | PHX_Substance_Inhalnt_1==1
         | PHX_Substance_Heroin_1==1
) %>%
  dplyr::filter(Age_At_PhenX_Completion > 14 | Age > 14, # age protocol violation
         SubjID != confusedID) ## that one confused drunk

Naives <- BehaveData %>%
  dplyr::filter( got_drunk == 0, 
           PHX_Tobacco_Status_1==2,
           PHX_Substance_LifeUse_5==2,  # marijuana
           PHX_Substance_Coc_1==2,    # crack or cocaine
           PHX_Substance_Stims_1==2,
           PHX_Substance_Meth_1==2,
           PHX_Substance_Sedtv_1==2,
           PHX_Substance_Tranq_1==2,
           PHX_Substance_Painkiller_1==2,
           PHX_Substance_Halluc_1==2,
           PHX_Substance_Inhalnt_1==2,
           PHX_Substance_Heroin_1==2  ) %>%
  dplyr::filter(Age_At_PhenX_Completion > 14 | Age > 14, # age protocol violation
         SubjID != confusedID) ## that one confused drunk

## this is just a quick check
## how many people said "Don't know / Decline to state" on any question?
Dunnos <- BehaveData %>%
  dplyr::filter( PHX_Tobacco_Status_1==99  ## don't know/won't state
          | PHX_Substance_LifeUse_5==99)  # marijuana don't 
dim(Dunnos)[1]  ## 8 people, OK, whatever
          
DruggieID <- Druggies %>%
  dplyr::select(  SubjID, 
           Alcohol = got_drunk,
           Tobacco = PHX_Tobacco_Status_1,
           Marijuana = PHX_Substance_LifeUse_5,  # marijuana
           Coca = PHX_Substance_Coc_1,
           Stims = PHX_Substance_Stims_1,
           Meth = PHX_Substance_Meth_1,
           Sedtv = PHX_Substance_Sedtv_1,
           Tranq = PHX_Substance_Tranq_1,
           Paink = PHX_Substance_Painkiller_1,
           Halluc = PHX_Substance_Halluc_1,
           Inhal = PHX_Substance_Inhalnt_1,
           Heroin = PHX_Substance_Heroin_1  )

NaiveID <- Naives %>%
  dplyr::select(  SubjID, 
           Alcohol = got_drunk,
           Tobacco = PHX_Tobacco_Status_1,
           Marijuana = PHX_Substance_LifeUse_5,  # marijuana
           Coca = PHX_Substance_Coc_1,
           Stims = PHX_Substance_Stims_1,
           Meth = PHX_Substance_Meth_1,
           Sedtv = PHX_Substance_Sedtv_1,
           Tranq = PHX_Substance_Tranq_1,
           Paink = PHX_Substance_Painkiller_1,
           Halluc = PHX_Substance_Halluc_1,
           Inhal = PHX_Substance_Inhalnt_1,
           Heroin = PHX_Substance_Heroin_1  )


intersect(Druggies,Naives)$SubjID  # should be zero
intersect(Druggies,Dunnos)$SubjID  # 4 people
intersect(Naives,Dunnos)$SubjID  # 0 people ... 
## that means 4 Dunnos were otherwise naive, and said 99 on Tobacco/MJ

# table(DruggieID$Tobacco,DruggieID$Alcohol) # 14 have smoked but never drink
smokers = subset(DruggieID, Tobacco == 1)
smokers = select(smokers, -Tobacco)
filter(smokers, Alcohol==0,
       Marijuana==2 | Marijuana==99,
       Coca==2, Stims==2, Meth==2,Sedtv==2,Tranq==2,Paink==2,
       Halluc==2, Inhal==2,Heroin==2)$SubjID  ### this is just 6 people



########### HERE ARE THE QUESTIONS
#----PHX_Alcohol_LifeAbuse_2,  --> became got_drunk (1=Ever, 0 =Never)
#"How old were you the first time you got drunk, that is, your speech was slurred or you were unsteady on your feet?"
#----PHX_Tobacco_Status_1,
#"Have you ever smoked part or all of a cigarette? (1=Yes, 2=No, 99=Don't Know/Decline to state)"
#----PHX_Substance_LifeUse_5,
#"In your entire life, have you EVER tried marijuana (pot, weed, hash, bud, doobie, reefer, mary-jane, puff)? (1=Yes, 2=No, 99=Don't know/Decline to State)
#----PHX_Substance_Coc_1,
#"Have you EVER used cocaine or crack? (1=Yes, 2=No)"
#----PHX_Substance_Stims_1,
#"Have you EVER used stimulants, for example, Preludin, Benzedrine, Methedrine, Ritalin, uppers, or speed? (1=Yes, 2=No)"
#----PHX_Substance_Meth_1,
#"Have you EVER used meth-amphetamines (crystal meth, ice, batu, crank, tine, tweak, glass, junk)? (1=Yes, 2=No)"
#----PHX_Substance_Sedtv_1,
#"Have you EVER used sedatives, for example, sleeping pills, barbiturates, Seconal, Quaaludes, or Chloral Hydrate? (1=Yes, 2=No)"
#----PHX_Substance_Tranq_1,
#"Have you EVER used tranquilizers or anti-anxiety drugs, for example, Valium, Librium, muscle relaxants, or Zanax (1=Yes, 2=No)"
#----PHX_Substance_Painkiller_1,
#"Have you EVER used painkillers, for example, Codeine, Darvon, Percodan, OxyContin, Dilaudid, Demerol, Celebrex, or Vioxx? (1=Yes, 2=No)"
#----PHX_Substance_Halluc_1,
#"Have you EVER used hallucinogens, for example, Ecstasy/MDMA, GHB, Ketamine, LSD, mescaline, psilocybin, PCP, angel dust, or peyote? (1=Yes, 2=No)
#----PHX_Substance_Inhalnt_1,
#"Have you EVER used inhalants or solvents, for example, amyl nitrite, nitrous oxide, glue, toluene or gasoline? (1=Yes, 2=No)
#----PHX_Substance_Heroin_1,
#"Have you EVER used heroin? (1=Yes, 2=No)"


#create the final dataset


Druggies$UseSubstance <-1
Naives$UseSubstance <-0




Data2 <- data.frame(rbind(Druggies,Naives))%>%
  left_join(GeneticData,by=c("SubjID")) %>%
  left_join(COMTData,by=c("SubjID")) 

Data <-Data2 %>%
  dplyr::select(GAF_africa,GAF_eastAsia,GAF_oceania,GAF_centralAsia,GAF_europe,GAF_amerind, FDH_3_Household_Income2,FDH_3_Household_IncomeLog,Gender,
         rs4680,rs12364283,rs1800497,
         PHX_PSS_TOTAL,
         PHX_IMP_TOTAL,
         Age_At_PhenX_Completion,
         UseSubstance
         )%>%
  na.omit()

table(Data$UseSubstance)



names(Data) <-c("Africa_Ancestry", "EastAsia_Ancestry","Oceania_Ancestry","CentralAsia_Ancestry","Europe_Ancestry","American_Ancestry","Household_Income","Log10_Household_Income","Gender","rs4680","rs12364283","rs1800497","Stress", "Impulsivity","Age","UseSubstance")
names(Data)

  

library(gmodels)
library(MASS)
options(xtable.comment = FALSE)
```

#Introduction
The high apparent hereditability of substance abuse disorders [XXXX]has long suggested that vulnerability may be genetic. There have now been broad studies comparing adult substance abusers to healthy controls in the hopes of identifying the genetic components of one’s risk. By identifying the specific genes that predict those who are affected by substance abuse disorders, we might be able to identify the mechanism underying various components of individuals' susceptibility to the drug in question. 

However, what is distinct about substance abuse is that one must first use a psychoactive substance to become addicted to that substance. Those who struggle with substance abuse appear to have lower discount rates (cite), and they are more risk seeking(cite) than the general population. Part of the genetic risk factors that may influence one’s likelihood of acquiring a debilitating dependence on a psychoactive substance may be those that influence the likelihood of trying or using the drug, despite the apparent adverse consequences.


However, the risky and behavior and possible poor judgement of those who try dangerous substances clearly does not exist in a vacuum. Our social and economic environment will influence both the availability of psychoactive substances, the messages we learn about the substances, and the magnitude of the consequences for using those substances. In adults there is a robust association between SES status and substance abuse (cite). However, there is signifcant debate about whether this is true for adolescents [@hanson2007socioeconomic,@goodman2002socioeconomic]. This suggest that, SES might not play a major role in likelihood of using a substance. Another possibility, however, is that SES might only have a protective influence for those who are particularly vulnerable. 

In this study we use a cognitive model proposed by @frank2007genetic that would let us identify the risk factors that might make it more likely that we could identify the underlying mechanisms in this model.

##Role of prefrontal dopamine density

@frank2007genetic found a genetic triple dissocaition between SNPs that regulated different components of the dopaminergic system. One component appeared to regulate prefronal, apparently model-based, control over decision making behavior in a reinforcement learning task. This was a functional single nucleotide polymorphism (SNP) involved in the COMT gene. Those with the MET allele have higher prefronal dopamine levels and greater top down control over learning.  A later paper by the same group of researchers [@doll2011dopaminergic] found that this same allele predicted the degree to which participants were likely to be influenced by explicit statements. Together this suggests that those with the MET allele, compared to those with the VAL allele that is associated with lower prefrontal dopamine function, might be more likely to internalize the explicit messages they receive about the dangers of using substances. Other researcher have started to look at whether variation in COMT function predicts susceptibiltiy to addition  [@m2012role], but the question of whether or not it predicts a greater likelihood of adolecents and younger adults trying drugs has not been studied.

##Role of striatal D2 receptor density

Another related SNP studied by these researchers was the C957T polymorphism of the DRD2 gene, that influenced participants ability to learn from negative reinforcement [@frank2007genetic]. DRD2 seems to have a selective effect on striatal, postsynaptic DA receptor density, and is implicated in learning from phasic dips in striatal dopamine. While the specific SNP used in that study, the C957T polymorphism is not available through PING, another SNP identified by @frank2009genetic, (rs12364283) influenced the same gene and appeared to have the same selective effecton on inhibitory learning. We hypothesized that those who are more likely to be able ot avoid negative outcomes may be more likely avoid use of dangergous psychoatctive substances.


#Methods

Data used in the preparation of this article were obtained from the Pediatric Imaging, Neurocognition and Genetics (PING) Study database (http://ping.chd.ucsd.edu/). PING was launched in 2009 by the National Institute on Drug Abuse (NIDA) and the Eunice Kennedy Shriver National Institute Of Child Health & Human Development (NICHD) as a 2-year project of the American Recovery and Reinvestment Act. The primary goal of PING has been to create a data resource of highly standardized and carefully curated magnetic resonance imaging (MRI) data, comprehensive genotyping data, and developmental and neuropsychological assessments for a large cohort of developing children aged 3 to 20 years. The scientific aim of the project is, by openly sharing these data, to amplify the power and productivity of investigations of healthy and disordered development in children, and to increase understanding of the origins of variation in neurobehavioral phenotypes. For up-to-date information, see http://ping.chd.ucsd.edu/.

##Participants   

Participants were recruited throuh PING, a nine site, cross-sectional study of roughly 1,400 children and adolencents from 3-20 years of age. Participants provided a saliva sample for whole genome sequencing, underwent three hours of neurocognitive testing and completed a one hour MRI session. Participants were subject to modest screening and those with a significant neurological disorder, or whose mother used alcohol or an illegal substance daily during pregnancy were screened.  Six of the nine sites participated in the additional testing, where children 14 and older were sent a web based interview of PhenX questionniares that included questions about their tobacco, alcohol and illegal substance use. We do not have access to the geographcial location of the participants. 



##Single Nucleotide Polymorphisms
Participants DNA was collected with a mouth swab and a whole genotype sequencing was performed using the Illumina Human660W-Quad BeadChip. Included in the 494,082 SNPs passing the PING quality control filters were the Val158Met polymorphism that influences the COMT gene (rs4680), and an SNP influencing post-synaptic D2 density (rs12364283) in the striatum (Frank & Hutchingson, 2011). For each participant, we assigened a scalar gene-dose value to them, definited in terms of the likely protective effect of their allels. For COMT, MET/MET =2, MET/VAL =0, VAL/VAL =-1, with a frequency of 
.  Their relative proportions are 14:81:56 respectively. For rs12364283, AA= 1, AT = -1 with a relative proportions of 141:10 respectively. Other studies have found a population freqency of roughly 15% for the AT allelic variety [@frank2009genetic], so the minority varient is significantly underrepresented in our population (Binomial test(exact) p = .002). The distribution of the rs12364283 and rs4680 allels apears to be significant $\chi^2$=0.232, p=.896.

```{r,results='hide',echo=FALSE}

binom.test(10,151,p=.15)

Data$COMT <- ifelse(Data$rs4680=="AA",1,NA)
Data$COMT <- ifelse(Data$rs4680=="AG",0,Data$COMT)
Data$COMT <- ifelse(Data$rs4680=="GG",-1,Data$COMT)

Data$DRD2 <-ifelse(Data$rs12364283=="AA",1,NA)
Data$DRD2 <-ifelse(Data$rs12364283=="GA",0,Data$DRD2)
# names(Data)

# xtable(table(Data$rs4680))
# # 
# xtable(table(Data$rs12364283))
# 
# xtable(table(Data$rs12364283,Data$rs4680))

#  chisq.test(table(Data$rs12364283,Data$rs4680))

```

To make sure this SNPs were not related to ethnic background, we includeded a measure of genetic ancestry for each participant. This was estimated by @akshoomoff2014nih, comparing PING participants to a large reference population. Each particpant was assigned a perentage ancestry to the six clusters which emerged from the data: Europe, Africa, East Asia, Oceania, Central Asia, American Indian To preven perfect colinearity we dropped the American Indian percentage. A Likelihood Ratio Test revealed that an that genetic ancessestry was boarderline significant effect of genetic ancestry on allelic type for the rs6480 SNP, $\chi^2$(5) =10.5, p = .06.  Similarly, at Likelihood Ratio Test revealed that there was a signifcant effect of ancestry on the rs12364283 allelic frequency $\chi^2$(5) =13, p = .02. Therefore, we will include genetic ancestry in our dataset, to control directly for the effect of genetic ancestry. 

```{r,echo=FALSE,results='asis'}

#names(Data)
g1 <- polr(as.factor(COMT)~Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry,data=Data,method="logistic")
g0 <-polr(as.factor(COMT)~1,data=Data,method="logistic")
anova(g0,g1,test="Chisq")

m1 <- (glm(DRD2~EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry,data=Data,family="binomial"))
m0 <- (glm(DRD2~1,data=Data,family="binomial"))

xtable(anova(m0,m1,test="LRT"))

```

## Family Income
Parents of participants were asked to indicate their family's total income in the past year.This was reported in a 12 point scale, with categories ranging from "<$5,000" to ">$300,000". To turn this data into a numeric veriable, we took the mean value for each category (i.e. "$20,000 to 29,999" was coded as 35,000). The endpoints were coded as the least extreme value(i.e. "20,000"), so that this coding scheme is a conservative estiamte of income variance. 



##Significant Substance Abuse. 
The PhenX (consensus measures for Phenotypes and eXposures) Toolkit (http://www.phenxtoolkit.org) was used to assess substance abuse. These standardized measures  were sponsored by the National Human Genome Research Institute to integrate behavioral outcomes and environmental factors with genomic data, and to facilitate cross-study comparisons(Hamilton et. al 2006) and multi-site investigations. The PING study administered PhenX as a web-based self-report questionnaire--either in-lab or at home--that took 20-90 minutes to complete, depending on age and substance exposure history. While our analysis focuses on items relating to substance use and addiction, PhenX measures of affective and mental health outcomes (emotional state, depression, anxiety, impulsivity, stress, self-esteem, self-efficacy) and school environment were also collected. Combined with the family history, cognitive testing, and genomic testing of each participant, this provided a rich dataset to address how SES might influence genetic contributions to substance use behavior. 
Our goal was to create a binary dependent variable that captured whether participants had engaged in high-risk substance abuse, so that our model would be predicting the probability of substance abuse. This is equivalent to creating two groups, Users and Naives. We reasoned that few participants would have developed full-blown addiction, especially in a pediatric sample, where parental vigilance, legal restrictions, and limited income reduces access to substances. Thus, classification as a User was based on ever having abused any substance included in the study.  We adopted this liberal criterion in the belief that becoming a User at a young age increases the risk of habit formation in adulthood, though our dataset cannot verify this.
In contrast, classification as Naive was more stringent, which allowed stronger claims to be tested. Because Naives had never consumed any drugs or gotten drunk, they may exhibit more inhibitory control and harm avoidance. Or these individuals may have reduced response to rewarding stimuli. Our model tests these hypotheses as group-level differences in the dopaminergic mechanisms described above. 
We recognize that some Naives, especially the younger ones, may choose to consume substances later in adolescence. Since genetic and household income influences are relatively stable, these mislabelled future-Users will weaken our analysis. This is a limitation of the PING design, since each participant completes PhenX at one timepoint in their lives. However, the inclusion of age as a predictor may mitigate this source of error.


*Age*. Participants younger than age 14 were not asked about alcohol and illicit substances; therefore, we excluded younger participants from our analysis. The age range (at the time of PhenX completion) was [14.5, 22.67] years.

*Substances*. Participants were classified as Users if they responded positively to ever having tried cigarettes, marijuana, cocaine, stimulants, meth-amphetamines, sedatives, tranquilizers, painkillers, hallucinogens, inhalants, or heroin--or if they had provided an age at which they first got drunk. (This was coded as a binary variable on ever having gotten drunk.) One participant was excluded for stating an age of first drunkenness that was younger than the age of initial alcohol use.
Participants were classified as Naives if they stated that they had never tried any of those substances, and had never gotten drunk. This includes ten individuals who had consumed more than a sip or taste of alcohol. 
Two questions (cigarettes and marijuana) allowed the response "Don't know/Decline to state". Eight participants selected this response on either question, but four of them already qualified as Users due to other substance history. The remaining four participants, who had no exposure to other drugs, were excluded from analysis, because they could not be confidently labelled as Naive.


#Results 

##Effecs of Age

Because this is a cross sectional study, we expect that age at the time of survey was likely to have a very large effect on the the likihood of using a substance. This is both because participants would have had more time to try a drug, and because earlier studies have shown older teens are more likely to have tried alcohol and other substances [@grant1997age]. To get a sense of this relationship, we plotted the proportion of users at each age who had used a psycho-active substance. Figure 1 shows this relationship.
```{r,echo=FALSE,fig.cap="Figure 1",fig.subcap="Error bars indicated 95% bionomial C.I."}

Data$AgeRound <- floor(Data$Age)
# names(Data)
PlotD <- Data %>%
  dplyr::group_by(AgeRound)%>%
  dplyr::summarize(x =as.numeric(sum(UseSubstance==1)),
            n =as.numeric(length(UseSubstance==1)))



binomest <-function(x,n){
  y= binom.test(x,n)
  return(list(y$estimate[1],y$conf.int[1],y$conf.int[2]))
}


PlotD$p <- 0
PlotD$plow <- 0
PlotD$phigh <- 1
for(i in 1:9){
  x= as.numeric(PlotD[i,"x"])
  n= as.numeric(PlotD[i,"n"])
  PlotD[i,"p"] <- binomest(x,n)[1]
  PlotD[i,"plow"] <- binomest(x,n)[2]
  PlotD[i,"phigh"] <- binomest(x,n)[3]
}


qplot(AgeRound,p,data=PlotD) + geom_errorbar(aes(ymin=plow,ymax=phigh,width=0)) +ylab("Probability of Using Substance")+xlab("Age") +ggtitle("Effect of Age on the Probability of Using a Substance")
```




##Ancestry

We were concerned that ancestry could explain the underlying genetic interactions through their association with the SNPs, we first tested the probability that that Ancestry either predicted substance use, or that ancestry interacted with log income to influence substance use. For both tests, we found no evidence that ancestry mattered $\chi^2$(4)=5.024, p = .28 and $\chi^2$(9)=10.917, p = .28. Therefore, we dropped ancestry from our later models.

```{r,results='hide',echo=FALSE}


m0 <- glm(UseSubstance~Age,family='binomial',data=Data)

m1 <- glm(UseSubstance~Age +Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry,family='binomial',data=Data)

m2 <- glm(UseSubstance~Age +Africa_Ancestry*Log10_Household_Income+
            EastAsia_Ancestry*Log10_Household_Income+
            EastAsia_Ancestry*Log10_Household_Income+
            CentralAsia_Ancestry*Log10_Household_Income+
            Europe_Ancestry*Log10_Household_Income,family='binomial',data=Data)

anova(m0,m1,test="Chisq")
anova(m0,m2,test="Chisq")


```


##Effect of Income and SNPs on Substance Use

We had three interacting hpothesis. The first is that the allelic variants that decreased COMT function and increased dopamine availablity in the prefrontal cortex would decrease the likelihood of using a substance. Second, that the allelic variants that increased DRD2 receptor density in the striatum would decrease the likelihood of using a substance. We put a greater focus on COMT as there was greater variablility in the genetic components. 



###COMT


```{r,echo=FALSE}
# qplot(FDH_3_Household_IncomeLog,UseSubstance,geom=c("point","smooth"),method = "lm",data=Data,facets=.~rs4680)+ylab("Predicted probability of using a su")


```

We conducted a logistic regression, the beta weights and Wald-z values are reported blow. As is clear, nothing except age is signifcant at the $\alpha$=.5 level. There is no evidence that the COMT polymorphism is related to substance use. 


```{r,echo=FALSE,results='asis'}
# m0 <- glm(UseSubstance~Age +Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry,family='binomial',data=Data)
# 
# m1 <-glm(UseSubstance~Age +Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry+COMT,family='binomial',data=Data)
# 
# # names(Data)
# m2<-glm(UseSubstance~Age+Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry+COMT+Log10_Household_Income,family='binomial',data=Data)
# 
# m3<-glm(UseSubstance~Age+Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry+COMT*Log10_Household_Income,family='binomial',data=Data)

m0 <- glm(UseSubstance~Age,family='binomial',data=Data)

m1 <-glm(UseSubstance~Age+COMT,family='binomial',data=Data)

# names(Data)
m2<-glm(UseSubstance~Age+COMT+Log10_Household_Income,family='binomial',data=Data)

m3<-glm(UseSubstance~Age+COMT*Log10_Household_Income,family='binomial',data=Data)


xtable(summary(m3))
```

However, as the likelihood ratio test below makes clear, there is no main effect of the COMT polymorphism.

```{r,echo=FALSE,results='asis'}
xtable(anova(m3,test="Chisq"))
```


###DRD2

We conducted a similar regression with DRD2. As you can see, the Wald-Z statistic is not significant, but there is a trend p<.01 towards an interaction. Because only 10 subjects, we are cautious in interpreting significance. However, the Wald Z test for the interaction is marginally significant (Wald-z=-1.73,p=.0824), and the $\chi^2$ likelihood ratio test is significant $\chi^2$(1)=5.56, p=0.02. 

```{r,echo=FALSE,results='asis'}

# m0 <- glm(UseSubstance~Age+Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry,family='binomial',data=Data)
# 
# m1 <-glm(UseSubstance~Age+Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry+DRD2,family='binomial',data=Data)
# 
# m2<-glm(UseSubstance~Age+Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry+DRD2+Log10_Household_Income,family='binomial',data=Data)
# 
# m3<-glm(UseSubstance~Age+Africa_Ancestry+EastAsia_Ancestry+EastAsia_Ancestry+CentralAsia_Ancestry+Europe_Ancestry+DRD2*Log10_Household_Income,family='binomial',data=Data)

m0 <- glm(UseSubstance~Age,family='binomial',data=Data)

m1 <-glm(UseSubstance~Age+DRD2,family='binomial',data=Data)

m2<-glm(UseSubstance~Age+DRD2+Log10_Household_Income,family='binomial',data=Data)

m3<-glm(UseSubstance~Age+DRD2*Log10_Household_Income,family='binomial',data=Data)




xtable(summary(m3))
```

```{r,echo=FALSE,results='asis'}
xtable(anova(m3,test="Chisq"))

```

However, it appeared to be related to age, those over 21, were significantly more likely to have the allele that hurts negative reinforcement learning. So we conducted a boostrapping, preserving all the current structure including the fact that 4 of the 10 people with the AG allel were over 21, and reran the Likelihood ratio test used above, preserving the at the significance test of the interaction. The p-value identified for our data  was smaller than 95.7% of the 10,000 bootstrapped samples, suggesting there is still boarderline evidence for a real effect. 

```{r,echo=FALSE,results='hide'}

Data3 <- Data

chisq.test(table(Data$Age>21,Data$DRD2),simulate.p.value = T)




sum(Data$Age>21)
 
Data3 %<>%
  dplyr::arrange(desc(Age))

# RandSampl <- function(Data3){
# Data3$NewDRD2<- c(sample(c(replicate(4,0),replicate(16,1)),20),sample(c(replicate(6,0),replicate(125,1)),131))
# m5 <- glm(UseSubstance~Age+NewDRD2*Log10_Household_Income,family='binomial',data=Data3)
# return(anova(m5,test="Chisq")[5,5])
# }

# Z <- replicate(10000,RandSampl(Data3))
# write.csv(Z, "z.csv")
Z<- read.csv("z.csv")
# qplot(Z$x)
eZ <- ecdf(Z$x)

actual <- anova(m3,test="Chisq")[5,5]

1-eZ(actual)
```


However, it appears that this might be very sensitive to just a few points, which should make us quite anxious about interpretation. Below the the values plotted, with a linear estimate their to ease interpretation (note we have used logistic regressions to estimate the true values). As should be apparent, those with the GA version of the rs12364283  did not span the whole range of household income, and the interaction could be driven by an outlier, which is made larger in the graph below. A conservative way of correcting for this would be to limit the range similarly for those with the AA allele, and removing the outlier, then rerunning the anova. It appears the that in by doing both, the relationship replain significant, allthough the interpretation appears more limited. It appears that for those who are making over $30,00 a year, there is some evidence that a higher income increases the likelihood of those wih the GA allele using drugs and alcohol. 

```{r,echo=FALSE}

Data$outlier <- 0.1
Data[122,"outlier"] <- 1
qplot(Household_Income,UseSubstance,geom=c("smooth"),method="glm",data=Data,facets=.~rs12364283)+geom_point(aes(Household_Income,jitter(UseSubstance),size=outlier))+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_continuous(labels = comma) 

qplot(Household_Income,UseSubstance,geom=c("smooth"),method="glm",data=Data[Data$Log10_Household_Income>4.5,],facets=.~rs12364283)+geom_point(aes(Household_Income,jitter(UseSubstance)))+ theme(axis.text.x = element_text(angle = 90, hjust = 1))+scale_x_continuous(labels = comma) 






# qplot(Log10_Household_Income,UseSubstance,geom=c("smooth"),method="lm",data=Data[c(-122),],facets=.~rs12364283,color=AgeRound)+geom_point(aes(Log10_Household_Income,jitter(UseSubstance)))

#data 136
# m4<-glm(UseSubstance~Age+DRD2*Log10_Household_Income,family='binomial',data=Data[c(-122),])
m5<-glm(UseSubstance~Age+DRD2*Log10_Household_Income,family='binomial',data=Data[Data$Log10_Household_Income>4.5,][-122,])

s <- Data[Data$Log10_Household_Income>4.5,][-122,]

```

```{r,results='asis',echo=FALSE}
xtable(anova(m5,test="Chisq"))


```

\pagebreak


#Discussion
It appears there is no strong evidence from this data that there is a significant inger

. This suggests that individual variation in D2 receptors does not significantly influence likelihood of using Tobacco, Alcohol or other illegal substances. This study is significantly limited by the ultimately very small sample of data that we had full data for, which limited our power. This dataset also had significantly fewer 





```{r,echo=FALSE,results='asis'}
# m0 <- glm(UseSubstance~Age,family='binomial',data=Data)
# 
# m1 <-glm(UseSubstance~Age+COMT+DRD2,family='binomial',data=Data)
# 
# m2 <- glm(UseSubstance~Age+COMT+DRD2+Log10_Household_Income,family='binomial',data=Data)
# 
# m3 <- glm(UseSubstance~Age+COMT+DRD2 +COMT*Log10_Household_Income+ DRD2*Log10_Household_Income,family='binomial',data=Data)
# # 
# # # Data$COMT <-ifelse(Data$rs4680=="AA",2,NA)
# # # Data$COMT <-ifelse(Data$rs4680=="AG",1,Data$COMT)
# # # Data$COMT <-ifelse(Data$rs4680=="GG",0,Data$COMT)
# # 
# # # m2 <- glm(COMT~Gender+log(Age_At_PhenX_Completion)+rs4680+rs12364283 +FDH_3_Household_IncomeLog,
# # #             family='binomial',data=Data)
# # 
# 
# # xtable(summary(m3))
# xtable(anova(m3,test="Chisq"))

```























#References

